sql/analyze.go

```diff
@@ -937,7 +937,24 @@ func simplifyComparisonExpr(n *parser.ComparisonExpr) parser.Expr {
 		}
 
 		switch n.Operator {
-		case parser.EQ, parser.NE, parser.GT, parser.GE, parser.LT, parser.LE:
+		case parser.EQ, parser.NE, parser.GE, parser.LE:
+			return n
+		case parser.GT:
+			// Note that if the variable is NULL, this would evaluate to NULL which
+			// would equivalent to false for a boolean expression. This
+			// simplification is necessary so that subsequent transformation of >
+			// constraint to >= can use Datum.Next without concern about whether a
+			// next value exists.
+			if n.Right.(parser.Datum).IsMax() {
+				return parser.DBool(false)
+			}
```

Comment:
I understand that you need this when `n.Right` is `parser.DInt(math.MaxInt64)`, but the part about `NULL` above confused me until I saw below that `DNull.IsMax()` is always true.